enumerables.rb

```diff
@@ -0,0 +1,121 @@
+module Enumerable
+    def my_each
+        if block_given?
+            item = 0
+            while item < self.size
+                yield(self[item])
+             item+=1
+            end
+        end
+    
+    self
+    end
+
+
+
+    def my_each_with_index
+        if block_given?
+            index = 0
+            while index < self.size
+                yield(self[index], index)
+                index+=1
+            end
+        end
+        self
+    end
+
+    def my_select
+        if block_given?
+            filtered = []
+            index = 0
+            self.my_each { |x| filtered.push(x) if yield(x) }
+        end
+        filtered
+    end
+    
+    def my_all?
+        if block_given?
+            not_all = 0
+            index = 0
+            while index < self.size
+                not_all = 1 if !yield(self[index])
+                index+=1
+                break if not_all == 1
+            end
+            
+        end
+        not_all == 1 ? false : true
+    end
+
+    def my_any?
+        if block_given?
+            any = 0
+            index = 0
+            while index < self.size
+                any = 1 if yield(self[index])
+                index+=1
+                break if any == 1
+            end
+        end
+        any == 1 ? true : false
+    end
+
+    def my_none?
+        if block_given?
+            some = 0
+            index = 0
+            while index < self.size
+                some = 1 if yield(self[index])
+                index+=1
+                break if some == 1
+            end
+        end
+        some == 1 ? false : true
+    end
```

Comment:
Please, see comment on `my_all?`

### Find more details about [`none?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F)

[OPTIONAL]

I think if we could get the block from `my_none?` to `my_any?` we should only need something like `!my_any?` for `my_none?` to work fine. For example:

```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg = nil,&block)
  end
```

To make this work we have to use `block.call(element)` instead of `yield(element)` in `my_any?`